atef/widgets/config/find_replace.py

```diff
@@ -0,0 +1,1009 @@
+            return False
+
+        return True
+
+
+class FindReplaceWidget(DesignerDisplay, QtWidgets.QWidget):
+
+    search_edit: QtWidgets.QLineEdit
+    replace_edit: QtWidgets.QLineEdit
+
+    case_button: QtWidgets.QToolButton
+    regex_button: QtWidgets.QToolButton
+
+    preview_button: QtWidgets.QPushButton
+    verify_button: QtWidgets.QPushButton
+    open_file_button: QtWidgets.QPushButton
+    open_converted_button: QtWidgets.QPushButton
+
+    change_list: QtWidgets.QListWidget
+
+    filename = 'find_replace_widget.ui'
+
+    def __init__(
+        self,
+        *args,
+        filepath: Optional[str] = None,
+        window: Optional[Any] = None,
+        **kwargs
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.fp = filepath
+        self._window = window
+        self.match_paths: Iterable[List[Any]] = []
+        self.orig_file = None
+
+        if not filepath:
+            self.open_converted_button.hide()
+        else:
+            self.open_file(filename=filepath)
+
+        if window:
+            self.open_converted_button.clicked.connect(self.open_converted)
+        else:
+            self.open_converted_button.hide()
+
+        self.setup_open_file_button()
+        self.preview_button.clicked.connect(self.preview_changes)
+        self.verify_button.clicked.connect(self.verify_changes)
+
+        self.replace_edit.editingFinished.connect(self.update_replace_fn)
+        self.search_edit.editingFinished.connect(self.update_match_fn)
+        # placeholder no-op functions
+        self._match_fn = lambda x: False
+        self._replace_fn = lambda x: x
+
+    def verify_changes(self) -> None:
+        verify_changes(self, self.orig_file)
+
+    def setup_open_file_button(self) -> None:
+        self.open_file_button.clicked.connect(self.open_file)
+
+    def open_file(self, *args, filename: Optional[str] = None, **kwargs) -> None:
+        if filename is None:
+            filename, _ = QtWidgets.QFileDialog.getOpenFileName(
+                parent=self,
+                caption='Select a config',
+                filter='Json Files (*.json)',
+            )
+        if not filename:
+            return
+
+        self.fp = filename
+        self.orig_file = self.load_file(filename)
+        self.setWindowTitle(f'find and replace: ({os.path.basename(filename)})')
+
+    def load_file(self, filepath) -> Union[ConfigurationFile, ProcedureFile]:
+        with open(filepath, 'r') as fp:
+            self._original_json = json.load(fp)
+        try:
+            data = deserialize(ConfigurationFile, self._original_json)
+        except ValidationError:
+            logger.debug('failed to open as passive checkout')
+            try:
+                data = deserialize(ProcedureFile, self._original_json)
+            except ValidationError:
+                logger.error('failed to open file as either active '
+                             'or passive checkout')
```

Comment:
This should probably `raise` and clear `original_json` 
At the very least `data` is unbound in this except clause